aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1747,6 +1546,78 @@ Tensor cholesky_inverse(const Tensor &input, bool upper) {
+                                   const Tensor& B,
+                                   bool left,
+                                   const Tensor& result,
+                                   const Tensor& LU,
+                                   const Tensor& pivots) {
+  // Possible optimization: Compute the LU factorization of A^T if A is contiguous
+  // Then we solve A^T X = B with adjoint=True
+  // This saves a copy as A doesn't need to be copied into an F-contig matrix in lu_factor
+  const bool use_A_T = A.is_contiguous() && !A.is_complex();
+  auto info = at::empty({0}, A.options().dtype(kInt));
+  at::linalg_lu_factor_ex_out(const_cast<Tensor&>(LU),
+                              const_cast<Tensor&>(pivots),
+                              const_cast<Tensor&>(info),
+                              use_A_T ? A.mT() : A,
+                              /*pivot=*/true,
+                              /*check_errors=*/false);
+  at::_linalg_check_errors(info, "torch.linalg.solve", A.dim() == 2);
+  at::linalg_lu_solve_out(const_cast<Tensor&>(result), LU, pivots, B, left, /*adjoint*/use_A_T);
+}
+
+Tensor& linalg_solve_out(const Tensor& A,
+                         const Tensor& B,
+                         bool left,
+                         Tensor& result) {
+
+  auto LU = at::empty({0}, A.options());
+  auto pivots = at::empty({0}, A.options().dtype(kInt));
+  at::_linalg_solve_out(result, LU, pivots, A, B, left);
+  return result;
+}
+
+// We implement linalg_solve as a composite function fo _linalg_solve
+Tensor linalg_solve(const Tensor& A,
+                    const Tensor& B,
+                    bool left) {
+  return std::get<0>(at::_linalg_solve(A, B, left));
+}
+
+std::tuple<Tensor&,Tensor&> solve_out(const Tensor& self, const Tensor& A, Tensor& solution, Tensor& lu) {
+  TORCH_WARN_ONCE(
+    "torch.solve is deprecated in favor of torch.linalg.solve",
+    "and will be removed in a future PyTorch release.\n",
+    "torch.linalg.solve has its arguments reversed and does not return the LU factorization.\n",
+    "To get the LU factorization see torch.linalg.lu_factor.\n",
+    "X = torch.solve(B, A).solution\n",
+    "should be replaced with\n",
+    "X = torch.linalg.solve(A, B)"
+  );
+  auto pivots = at::empty({0}, A.options().dtype(kInt));
+  at::_linalg_solve_out(solution, lu, pivots, A, self);
+  return std::tie(solution, lu);
+}
+
+std::tuple<Tensor,Tensor> solve(const Tensor& self, const Tensor& A) {
+  TORCH_WARN_ONCE(
+    "torch.solve is deprecated in favor of torch.linalg.solve",
+    "and will be removed in a future PyTorch release.\n",
+    "torch.linalg.solve has its arguments reversed and does not return the LU factorization.\n",
+    "To get the LU factorization see torch.linalg.lu_factor.\n",
+    "X = torch.solve(B, A).solution\n",
+    "should be replaced with\n",
+    "X = torch.linalg.solve(A, B)"
+  );
+  Tensor X, LU, pivots;
+  std::tie(X, LU, pivots) = at::_linalg_solve(A, self);
```

Comment:
This also enables the use of 1D `self` in the `torch.solve` call. Maybe it's fine.

Reply:
What do you mean by "maybe it's fine"?

Reply:
This function is deprecated and will be removed eventually so maybe we can ignore maintaining its documentation up-to-date. Before this PR the following would raise an error:
```py
import torch
a = torch.randn(3, 3)
b = torch.randn(3)
res = b.solve(a)
# RuntimeError: B should have at least 2 dimensions, but has 1 dimensions instead
```
This with PR it would work, but the documentation says that `b` needs to be at least 2D.

Reply:
I see what you mean. At least it's not BC-breaking, and perhaps makes writing the update rule easier, as we're stating that this function is just a special case of `linalg.solve`?
Now, if you think this is not the case, I can revert those changes and leave the removal of this function to your PR. Also fine with me.

Reply:
Let's revert any changes to `torch.solve` and keep this PR slimmer.

Reply:
We don't need to write the update rule, you've done this job already and it's in the docs. https://pytorch.org/docs/stable/generated/torch.solve.html